llama-index-core/llama_index/core/query_pipeline/query.py

```diff
@@ -438,12 +559,34 @@ def _run(self, *args: Any, return_values_direct: bool = True, **kwargs: Any) ->
 
         """
         root_key, kwargs = self._get_root_key_and_kwargs(*args, **kwargs)
-        # call run_multi with one root key
-        result_outputs = self._run_multi({root_key: kwargs})
-        return self._get_single_result_output(result_outputs, return_values_direct)
+
+        if show_intermediate_outputs:
+            result_outputs, intermediates = self._run_multi(
+                {root_key: kwargs}, show_intermediate_outputs=show_intermediate_outputs
+            )
+            formatted_intermediates = {
+                key: [
+                    self._get_multiple_result_output(value, return_values_direct),
+                    value,
+                ]
+                for key, value in intermediates.items()
+            }
+            return (
```

Comment:
Ok! Sorry for the delay on getting back to this -- I wanted to take some time to step through the code in a debugger and see what was going on

So, we return a tuple of (normal_output, Dict[(output + input pairs)]) I think?

This structure was a little hard to to parse -- I wonder if we should introduce a specific output object for this instead?  Something like

```
class ModuleInputOutputs:
  inputs: Dict  # using input_keys ?
  outputs: Dict # using output_keys ?
  
  
intermediate_results = {"module_name": ModuleInputOutputs(inputs={..}, outputs={...}), ..}
```

Thoughts?

Either way, we should also update some of the docs to show off this feature 💪🏻

Reply:
Sure that sounds good!

Reply:
I'll update the docs too

Reply:
@xpbowler Wow, I saw this comment one hour too late 😅 Just pushed a change for what I was thinking, whoops. Feel free to revert!

Reply:
This looks good to me, I think it is a nice interface!

Reply:
updated docs